tests/e2e/tests/product-gallery/inner-blocks/product-gallery-large-image-next-previous/product-gallery-large-image-next-previous.block_theme.spec.ts

```diff
@@ -0,0 +1,327 @@
+	test.beforeEach( async ( { requestUtils, admin, editorUtils } ) => {
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+		await admin.visitSiteEditor( {
+			postId: `woocommerce/woocommerce//${ blockData.slug }`,
+			postType: 'wp_template',
+		} );
+		await editorUtils.enterEditMode();
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+
+	test( 'Renders Next/Previous Button block on the editor side', async ( {
+		editorUtils,
+		editor,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'woocommerce/product-gallery',
+		} );
+
+		const block = await editorUtils.getBlockByName( blockData.name );
+
+		await expect( block ).toBeVisible();
+	} );
+
+	test( 'Renders Next/Previous Button block on the frontend side', async ( {
+		admin,
+		editorUtils,
+		frontendUtils,
+		editor,
+		page,
+	} ) => {
+		await addBlock( admin, editor, editorUtils );
+
+		await Promise.all( [
+			editor.saveSiteEditorEntities(),
+			page.waitForResponse( ( response ) =>
+				response.url().includes( 'wp-json/wp/v2/templates/' )
+			),
+		] );
+
+		await page.goto( blockData.productPage, {
+			waitUntil: 'commit',
+		} );
+
+		const block = await frontendUtils.getBlockByName( blockData.name );
+
+		await expect( block ).toBeVisible();
+	} );
+
+	test.describe( `${ blockData.name } Settings`, () => {
+		test( 'Hide correctly the arrows', async ( {
+			page,
+			editor,
+			editorUtils,
+			admin,
+		} ) => {
+			await addBlock( admin, editor, editorUtils );
+			await (
+				await editorUtils.getBlockByName( blockData.name )
+			 ).click();
+			await editor.openDocumentSettingsSidebar();
+			await page
+				.locator( blockData.selectors.editor.noArrowsOption )
+				.click();
+
+			const isVisible = await page
+				.locator(
+					'.wc-block-product-gallery-large-image-next-previous-container'
+				)
+				.isVisible();
+
+			expect( isVisible ).toBe( false );
+
+			await editor.saveSiteEditorEntities();
+
+			await page.goto( blockData.productPage, {
+				waitUntil: 'commit',
+			} );
+
+			const leftArrow = await page
+				.locator( blockData.selectors.editor.leftArrow.off )
+				.isVisible();
+
+			const rightArrow = await page
+				.locator( blockData.selectors.editor.rightArrow.off )
+				.isVisible();
+
+			expect( leftArrow ).toBe( false );
+			expect( rightArrow ).toBe( false );
+		} );
+
+		test( 'Show button outside of the image', async ( {
+			page,
+			editor,
+			editorUtils,
+			frontendUtils,
+		} ) => {
+			// Currently we are adding the block under the related products block, but in the future we have to add replace the product gallery block with this block.
+			const parentBlock = await editorUtils.getBlockByName(
+				'woocommerce/product-image-gallery'
+			);
+			const clientId =
+				// eslint-disable-next-line playwright/no-conditional-in-test
```

Comment:
I don't think it's a good idea to ignore this eslint rule. Can you think of any scenario where it's acceptable for the parent clientId to be undefined?